GitHubA11yFixes.user.js

```diff
@@ -15,6 +15,24 @@ function makeHeading(elem, level) {
 	elem.setAttribute("aria-level", level);
 }
 
+var idCounter = 0;
+// Get a node's id. If it doesn't have one, make and set one first.
+function setAriaIdIfNecessary(elem) {
+	if (elem.id) {
+		return elem.id;
+	}
+	elem.setAttribute("id", "axsg-" + idCounter++);
+	return elem.id;
+}
+
+function makeElementOwn(parentElement, listOfNodes){
+	ids = [];

```

Comment:
nit: This will end up as a global because the scope isn't restricted. Please add a "var" keyword.